doc/tasks/readme.md

```diff
@@ -0,0 +1,737 @@
+  Spielendenaktion (bspw. Tastendruck) aufgehoben werden soll
+
+### Spielszenario
+
+Ein Spielszenario beschreibt, welche Spielmechaniken, welche Entitätstypen und auf welche
+Art diese in einem konkreten Dungeonlevel in Verbindung gebracht werden, um eine abstrakte
+Aufgabe zu realisieren. Ein Spielszenario beschreibt nicht, welche visuellen
+
+Beispiel: In einem Raum befindet sich ein NPC und verschiedene Schalter. Der NPC stellt dem
+Spieler eine Frage und sagt ihm, welcher Hebel für welche Antwort steht.
+
+### Steuermechanismen
+
+Steuermechanismen sind alle Mechanismen des Systems, die den Ablauf der Umsetzung einer
+Aufgabe in ein Spielszenario beeinflussen.
+
+Steuermechanismen grenzen sich von Spielmechaniken ab, indem sie die übergeordneten
+Mechanismen realisieren, die sich nicht ausschließlich auf Game-Design Elemente beziehen.
+Hierzu zählen unter anderem das Logging von Antworten, die Organisation der Aufgabenfolge
+und Aufgabenverschachtelung.
+
+## Aufgabentypen
+
+### Single-Choice
+
+Die Aufgabenstellung besteht aus einem Fragetext und einer Menge an vorgegebenen
+Antwortmöglickeiten. Nur eine Antwort ist richtig. Die richtige Antwort muss ausgewählt
+werden.
+
+### Multiple-Choice
+
+Die Aufgabenstellung besteht aus einem Fragetext und einer Menge an vorgegebenen
+Antwortmöglickeiten. Mehrere Antworten sind richtig. Es müssen alle richtigen Antworten
+markiert werden. Es dürfen keine falschen Antworten markiert werden. Die Reihenfolge spielt
+keine Rolle.
+
+### Lückentext
+
+Die Aufgabenstellung besteht aus einem Fragetext. Die Antwort besteht aus einem oder
+mehreren Freitexten. Es muss jeweils die richtige Antwort in den Freitext-Stellen eingegeben
+werden. Die richtige Antwort kann ein konkreter Wert oder eine Eingabe passend zu einem
+regulären Ausdruck sein.
+
+### Ersetzen
+
+Aus einer Gesamtmenge aus $l$ Elementen (mit $0 < l$) muss eine Menge mit $m$ Elementen (mit $0 < n$) durch eine
+zweite Menge mit $m$ Elementen (mit $0 < m$) ersetzt werden.
+Hierdurch wird die Gesamtmenge der Elemente sowie die Anzahl der enthaltenen Elemente verändert.
+
+### Zuordnen
+
+Zuordnen bezeichnet im Allgemeinen die Aufgabe, Elemente aus einer Menge $A$ (z.B.
+Antwortmöglichkeiten) die Elemente aus einer anderen Menge $B$ zuzuordnen (bspw. Array-Index,
+Reihe-Spalte Kombination einer Matrix). Hierbei muss nicht jedes Element aus $A$ der
+Antwortmöglichkeiten auch einem Element aus $B$ zugeordnet werden. Allerdings muss jedem Element aus $B$ ein
+Element aus $A$ zugeordnet werden. Die gesamte Menge $A$ der Antwortmöglichkeiten muss den Studierenden
+bekannt sein, bevor sie den ersten Zuordnungsschritt machen müssen.
+
+### Elemente in der richtigen Reihenfolge auswählen
+
+Es muss in der richtigen Reihenfolge ein Element nach dem anderen aus einer Menge $A$ ausgewählt
+werden.
+Den Studierenden müssen nicht alle Elemente aus $A$ bekannt sein, bevor sie das erste Element
+auswählen müssen.
```

Comment:
Das verstehe ich nicht. Wenn es auf die Reihenfolge ankommt, muss ich doch dann mindestens das jeweils nächste Element kennen?

Reply:
Das stimmt, aber eben nicht alle. Der Satz soll insbesondere zur Abgrenzung ggü. Zuordnen dienen. Da das jetzt schon so häufig zu Verwirrung geführt hat, werde ich nochmal ein Szenario dafür schreiben, was das hoffentlich klarer macht. 

Das Beispiel was mir da immer einfällt, ist das Auswählen eines Elements, indem Türen in der richtigen Reihenfolge wählt. Angenommen, dass es insgesamt 9 Elemente in $A$ gibt, und davon 3 in der richtigen Reihenfolge gewählt werden müssen. Dann wäre das Szenario, dass in 3 **aufeinanderfolgenden** Räumen jeweils die richtige aus 3 Türen (eine pro Antwortmöglichkeit)  gewählt werden muss. Die Studierenden haben dann immer nur Zugriff auf 3 der 9 Elemente und nie auf alle 9.

Dieses Szenario könnte man natürlich auch durch drei aufeinanderfolgende Single Choice Fragen abbilden, allerdings haben wir das eigentlich schon entschieden, dass das hier trotzdem ein eigenständiger Aufgabentyp sein soll.

Dass nicht alle Elemente aus $A$ gewählt werden müssen fehlt noch in der Beschreibung, fällt mir gerade noch auf. Das würde ich auch noch ergänzen, falls wir uns jetzt entscheiden, den Aufgabentyp zu behalten. @AMatutat, @AKirsch1, @bcg7 Meinungen hierzu?

Reply:
> Dann wäre das Szenario, dass in 3 aufeinanderfolgenden Räumen jeweils die richtige aus 3 Türen (eine pro Antwortmöglichkeit) gewählt werden muss.

@malt-r Das hat doch nichts mit "in der richtigen Reihenfolge auswählen", sondern das wären für mich 3 aufeinanderfolgende Aufgaben vom Typ "Single Choice"? 

Ich bin nach wie vor der Meinung, dass ich die Elemente kennen muss, die ich in der richtigen Reihenfolge auswählen muss. Und eigentlich ist "Multiple Choice" nur ein Spezialfall von "in der richtigen Reihenfolge auswählen" (bei MC fällt nur die Reihenfolge weg).

Reply:
> Und eigentlich ist "Multiple Choice" nur ein Spezialfall von "in der richtigen Reihenfolge auswählen" (bei MC fällt nur die Reihenfolge weg).


So hab ich auch gedacht. Nur dann kommst du eigentlich darauf, dass Multiple Choice nur eine Kette an Single-Choice fragen ist.

Reply:
So wie ich das sehe haben wir hier zwei Möglichkeiten:
1. Wir diskutieren *nochmal* im nächsten Mittwoch über die Rechtfertigung des Aufgabentyps und das bleibt noch länger liegen, obwohl das eigentlich schon entschieden war (das Beispiel mit den Türen habe ich auch schon unter https://github.com/Programmiermethoden/Dungeon/issues/519 angeführt).
2. Wir handeln im Sinne des Pragmatismus und streichen den Aufgabentyp. Ich kann das Argument, dass das Beispiel eigentlich 3 SC Fragen sind, gut nachvollziehen. Wenn zusätzlich die Einschränkung, dass nicht alle Elemente vorher bekannt sein müssen, wegfällt, dann grenzt sich der Aufgabentyp meiner Meinung nach zu schwach von "Zuordnen" ab (was auch hier: https://github.com/Programmiermethoden/2022-freiraum-dungeon/pull/110#discussion_r1147954617 schonmal Diskussionsgegenstand war)

Reply:
Ja, im Prinzip könnte man MC als Folge von SC-Aufgaben modellieren. Dann hätte man eben mehrere Aufgaben statt einer.

Wenn man aber über _eine_ Aufgabe nachdenkt, dann ist MC ein Spezialfall von "in richtiger Reihenfolge auswählen". Das funktioniert sogar in verschiedenen Repräsentationen, beispielsweise mit Anhaken von Checkboxen oder Öffnen von Kisten oder Aufnehmen oder Ablegen von Items ...

Reply:
> So wie ich das sehe haben wir hier zwei Möglichkeiten:
> 
> 1. Wir diskutieren _nochmal_ im nächsten Mittwoch über die Rechtfertigung des Aufgabentyps und das bleibt noch länger liegen, obwohl das eigentlich schon entschieden war (das Beispiel mit den Türen habe ich auch schon unter [[AUFGABEN] Merge von Aufgabentypen "Positionieren" und "Elemente in der richtigen Reihenfolge wählen" #519](https://github.com/Programmiermethoden/Dungeon/issues/519) angeführt).
> 2. Wir handeln im Sinne des Pragmatismus und streichen den Aufgabentyp. Ich kann das Argument, dass das Beispiel eigentlich 3 SC Fragen sind, gut nachvollziehen. Wenn zusätzlich die Einschränkung, dass nicht alle Elemente vorher bekannt sein müssen, wegfällt, dann grenzt sich der Aufgabentyp meiner Meinung nach zu schwach von "Zuordnen" ab (was auch hier: [Kontrollmechanismus Brainstorming 2022-freiraum-dungeon#110 (comment)](https://github.com/Programmiermethoden/2022-freiraum-dungeon/pull/110#discussion_r1147954617) schonmal Diskussionsgegenstand war)

Naja, es zeigt sich halt in dieser Diskussion, dass die Konzepte offenbar doch noch nicht so klar oder gut abgegrenzt waren/sind, wie wir alle gedacht hatten. 

Deshalb würde ich den Typ aber nicht gleich sofort streichen - man muss halt nochmal nachdenken. Ist "in richtiger Reihenfolge auswählen" eventuell ein Spezialfall von "Zuordnen"?

Meeting: Entweder wir lassen diesen PR offen und diskutieren das im nächsten Meeting, oder es gibt ein Issue für diese Diskussion und dann kann hier resolved werden und entsprechend gemergt werden und das Issue im nächsten Meeting besprochen werden.

Reply:
> So wie ich das sehe haben wir hier zwei Möglichkeiten:
> 
> 1. Wir diskutieren _nochmal_ im nächsten Mittwoch über die Rechtfertigung des Aufgabentyps und das bleibt noch länger liegen, obwohl das eigentlich schon entschieden war (das Beispiel mit den Türen habe ich auch schon unter [[AUFGABEN] Merge von Aufgabentypen "Positionieren" und "Elemente in der richtigen Reihenfolge wählen" #519](https://github.com/Programmiermethoden/Dungeon/issues/519) angeführt).
> 2. Wir handeln im Sinne des Pragmatismus und streichen den Aufgabentyp. Ich kann das Argument, dass das Beispiel eigentlich 3 SC Fragen sind, gut nachvollziehen. Wenn zusätzlich die Einschränkung, dass nicht alle Elemente vorher bekannt sein müssen, wegfällt, dann grenzt sich der Aufgabentyp meiner Meinung nach zu schwach von "Zuordnen" ab (was auch hier: [Kontrollmechanismus Brainstorming 2022-freiraum-dungeon#110 (comment)](https://github.com/Programmiermethoden/2022-freiraum-dungeon/pull/110#discussion_r1147954617) schonmal Diskussionsgegenstand war)

@malt-r Ich kann Deinen Widerwillen verstehen, aber es zeigt sich halt in dieser Diskussion, dass die Konzepte offenbar doch noch nicht so klar oder gut abgegrenzt waren/sind, wie wir alle gedacht hatten. 

Deshalb würde ich den Typ aber nicht gleich sofort streichen - man muss halt nochmal nachdenken. Ist "in richtiger Reihenfolge auswählen" eventuell ein Spezialfall von "Zuordnen"?

Meeting: Entweder wir lassen diesen PR offen und diskutieren das im nächsten Meeting, oder es gibt ein neues Issue für diese Diskussion und dann kann hier resolved werden und entsprechend gemergt werden und das Issue im nächsten Meeting besprochen werden.